src/content/docs/en/guides/cms/hashnode.mdx

```diff
@@ -0,0 +1,352 @@
+---
+title: Hashnode & Astro
+description: Add content to your Astro project using Hashnode as a CMS
+type: cms
+stub: false
+service: Hashnode
+i18nReady: true
+---
+
+import { FileTree } from '@astrojs/starlight/components';
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+[Hashnode](https://hashnode.com/) is a hosted CMS that allows you to create a blog or publication.
+
+## Integrating with Astro
+The [Hashnode Public API](https://apidocs.hashnode.com/) is a GraphQL API that allows you to interact with Hashnode. This guide uses [`graphql-request`](https://github.com/jasonkuhrt/graphql-request), a minimal GraphQL client that works well with Astro, to bring your Hashnode data into your Astro project.
+
+### Prerequisites
+To get started you will need to have the following: 
+
+1. **An Astro project** - If you don't have an Astro project yet, our [Installation guide](/en/install/auto/) will get you up and running in no time.
+
+2. **A Hashnode site** - You can create free personal site by visiting [Hashnode](https://hashnode.com/).
+
+### Installing dependencies
+
+Install the `graphql-request` package using the package manager of your choice:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+  ```shell
+  npm install graphql-request
+  ```
+  </Fragment>
+  <Fragment slot="pnpm">
+  ```shell
+  pnpm add graphql-request
+  ```
+  </Fragment>
+  <Fragment slot="yarn">
+  ```shell
+  yarn add graphql-request
+  ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Fetching data
+
+Can we have a minimal example here?
+
+## Making a blog with Astro and Hashnode
+
+This guide uses [`graphql-request`](https://github.com/jasonkuhrt/graphql-request), a minimal GraphQL client that works well with Astro, to bring your Hashnode data into your Astro project.
+
+### Prerequisites
+
+1. A Hashnode Blog 
+2. An Astro project integrated with the [graphql-request](https://github.com/jasonkuhrt/graphql-request) package installed.
+
+This example will create an index page that lists posts with links to dynamically-generated individual post pages.
+
+### Fetching Data
+
+1. To fetch your site's data with the `graphql-request` package, make a `src/lib` directory and create two new files `client.ts` & `schema.ts` files:
+
+<FileTree title="Project Structure">
+- src/
+  - lib/
+    - **client.ts**
+    - **schema.ts**
+  - pages/
+    - index.astro
+- astro.config.mjs
+- package.json
+</FileTree>
+
+2. Initialize an API instance with the GraphQLClient using the URL from your Hashnode Website.
+
+```ts title="src/lib/client.ts"
+
+import { gql, GraphQLClient } from "graphql-request";
+import type { AllPostsData, PostData } from "./schema";
+
+export const getClient = () => { 
+  return new GraphQLClient("https://gql.hashnode.com") 
+}
+
+const myHashnodeURL = "astroplayground.hashnode.dev";
```

Comment:
How about highlighting the the HashnodeURL to readers ( I skimmed over that during my walkthrough)?
```suggestion
```ts title="src/lib/client.ts" "astroplayground.hashnode.dev"

import { gql, GraphQLClient } from "graphql-request";
import type { AllPostsData, PostData } from "./schema";

export const getClient = () => { 
  return new GraphQLClient("https://gql.hashnode.com") 
}

const myHashnodeURL = "astroplayground.hashnode.dev";
```

Reply:
I like that suggestion... lol i always forget you can do that